src/video/video_source_redirect.rs

```diff
@@ -0,0 +1,93 @@
+use super::types::*;
+use super::video_source::{VideoSource, VideoSourceAvailable};
+
+use paperclip::actix::Apiv2Schema;
+use serde::{Deserialize, Serialize};
+
+#[derive(Apiv2Schema, Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub enum VideoSourceRedirectType {
+    Redirect(String),
+}
+
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub struct VideoSourceRedirect {
+    pub name: String,
+    pub source: VideoSourceRedirectType,
+}
+
+impl VideoSource for VideoSourceRedirect {
+    fn name(&self) -> &String {
+        return &self.name;
+    }
+
+    fn source_string(&self) -> &str {
+        match &self.source {
+            VideoSourceRedirectType::Redirect(string) => &string,
+        }
+    }
+
+    fn formats(&self) -> Vec<Format> {
+        match &self.source {
+            VideoSourceRedirectType::Redirect(_) => {
+                vec![Format {
```

Comment:
We could just make an empty vector, no ?

Reply:
Wait, this is masked as solved, but we could still be using an empty vector here.
What is the point ?